bowling/src/input.cpp

```diff
@@ -0,0 +1,65 @@
+#include <algorithm>
+#include <fstream>
+
+#include "input.hpp"
+
+namespace fs = std::experimental::filesystem;
+
+Input::Input(const std::string& directory) : directoryPath_(fs::path(directory)), isValid_(false) {
+    if (!fs::exists(directoryPath_)) {
+        return;
+    }
+    if (!fs::is_directory(directoryPath_)) {
+        return;
+    }
+    std::vector<fs::path> files = getFiles();
+    readLanes(files);
+    lanesNum_ = lanes_.size();
+    isValid_ = true;
+};
+
+std::vector<fs::path> Input::getFiles() {
+    std::vector<fs::path> files;
+
+    for (const auto& entry : fs::directory_iterator(directoryPath_)) {
+        auto filename = entry.path().filename();
+        if (fs::is_regular_file(entry.status())) {
+            files.push_back(entry.path());
+        }
+    }
+    files.shrink_to_fit();
+    std::sort(files.begin(), files.end());
+
+    return files;
+}
+
+std::shared_ptr<Lane> Input::getLane(size_t index) {
+    if (index < lanesNum_) {
+        return lanes_[index];
+    }
+    return nullptr;
+}
+
+void Input::readLanes(const std::vector<fs::path>& files) {
```

Comment:
maybe new Class for it?  like `laneFileReader` 🤔 
to consider

Reply:
overkill to me

Reply:
But it will be more readable

Reply:
Less readable for me